pycares/_cfficore/__init__.py

```diff
@@ -477,15 +477,15 @@ def process_fd(self, read_fd, write_fd):
         _lib.ares_process_fd(self.channel, _ffi.cast("ares_socket_t", read_fd), _ffi.cast("ares_socket_t", write_fd))
 
     @check_channel
-    def timeout(self, t = -1):
+    def timeout(self, t = None):
         maxtv = _ffi.NULL
         tv = _ffi.new("struct timeval*")
 
         if t >= 0.0:
             maxtv = _ffi.new("struct timeval*")
             maxtv.tv_sec = int(math.floor(t))
             maxtv.tv_usec = int(math.fmod(t, 1.0) * 1000000)
-        elif t == -1:
+        elif t in (None, -1):
```

Comment:
you will need to reverse this test, because None can no longer be compared to floats on some Python versions

Reply:
Do you recall Python's version so I could test against it?

Reply:
You can check on the Travis CI output, but AFAIS Python 3.6 has this.

Reply:
Perhaps I'm thinking of something else, but
```
>>> 1.0 == None
False
```
with 3.6.3